### PR TITLE
GEODE-6833: Updating P2P SSL auth

### PR DESCRIPTION
**Pull request by echobravopapa**
Added test using 2 servers, no clients and associated keystone/truststore files.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x ] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    r.put("serverkey", "servervalue");
+  }
+
+  private void doServerRegionTest() {
+    Region<String, String> region = cache.getRegion("serverRegion");
+    assertEquals("servervalue", region.get("serverkey"));
+  }
+
+  private static Integer setUpLocatorTask() throws Exception {
+    return instance.setUpLocator();
```
Comment (mcmellawatt) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: Do single line helper methods buy us much?  Same comment on the next few methods.

File: geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java [hunk 2]
```diff
@@ -1835,6 +1837,8 @@ private void createIoFilter(SocketChannel channel, boolean clientSocket) throws
       if (!clientSocket) {
         engine.setWantClientAuth(true);
         engine.setNeedClientAuth(true);
+//        engine.setNeedClientAuth(SSLConfigurationFactory.getSSLConfigForComponent(getConduit().config,
```
Comment (mcmellawatt) [inline geode-core/src/main/java/org/apache/geode/internal/tcp/Connection.java]: Commented out code

File: geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      keyStore =
+          createTempFileFromResource(SSLDualServerNoClientAuthDUnitTest.class, SERVER_2_KEYSTORE)
+              .getAbsolutePath();
+      trustStore =
+          createTempFileFromResource(SSLDualServerNoClientAuthDUnitTest.class,
+              SERVER_2_TRUSTSTORE)
+                  .getAbsolutePath();
+    }
+    gemFireProps.setProperty(SSL_ENABLED_COMPONENTS, "cluster");
+    gemFireProps.setProperty(SSL_REQUIRE_AUTHENTICATION, "" + cacheServerSslRequireAuth);
```
Comment (mcmellawatt) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: Again, concatenating with empty string...why?

File: geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+    String cacheServerSslprotocols = "any";
+    String cacheServerSslciphers = "any";
+    boolean cacheServerSslRequireAuth = false;
+
+    System.setProperty("javax.net.debug", "all");
+    String keyStore;
+    String trustStore;
+    if (VM.getCurrentVMNum() == 1) {
+      keyStore =
```
Comment (mcmellawatt) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: Your call, but you could extract the logic in this block into a helper method that takes the keystore and truststore as parameters and reduce duplication with what is in the else block

File: geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    serverVM.invoke(() -> setUpServerVMTask(locatorPort));
+    server2VM.invoke(() -> setUpServerVMTask(locatorPort));
+
+    server2VM.invoke(() -> doServerRegionTestTask());
+    serverVM.invoke(() -> doServerRegionTestTask());
+  }
+
+  private void createCache(Properties props) throws Exception {
+    cache = new CacheFactory(props).create();
+    if (cache == null) {
```
Comment (mcmellawatt) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: Does this actually ever happen in practice?

File: geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+      throw new Exception("CacheFactory.create() returned null ");
+    }
+  }
+
+  private Integer setUpLocator() throws Exception {
+    Properties gemFireProps = new Properties();
+
+    String cacheServerSslprotocols = "any";
```
Comment (mcmellawatt) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: Could be final - I saw some other similarly un-mutated strings declared final so we should at least try and be consistent.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+    Locator.startLocatorAndDS(0, new File(""), gemFireProps);
+
+    return Locator.getLocator().getPort();
+  }
+
+  private void setUpAndConnectToDistributedSystem(Integer locatorPort) throws Exception {
+    Properties gemFireProps = new Properties();
+
+    String cacheServerSslprotocols = "any";
```
Comment (mcmellawatt) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: final

Comment (kirklund) [reply geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: According to Effective Java 3rd Edition, all local variables that are set only once are implicitly made `final` by the JVM without requiring use of the keyword. So it really just becomes noise (one more word on many lines). The only place where it's valuable is on class fields and constants.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java (hunk 4 above)
Comment (echobravopapa) [reply geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java]: there is dual output so simple Method extraction is not available and in this case a Method object seems overkill

**Pull request merged**